### PR TITLE
22 implement transpose array operation

### LINKED ISSUE #22: Implement transpose Array Operation
### Description
Implement array transpose operation in `org.fungover.breeze.util.Arrays` to convert rows to columns and vice versa.

### Implementation Requirements
```java
public static <T> T[][] transpose(T[][] array)
```
- Transposes 2D array
- Validates array regularity
- Handles empty arrays
- Preserves type information

### Test Cases
- Square arrays
- Rectangular arrays
- Single row/column arrays
- Empty array
- Irregular arrays (should throw)
- Large arrays (performance)
- Arrays with null elements

### Acceptance Criteria
- [ ] Method implemented
- [ ] 100% test coverage
- [ ] Performance benchmarks
- [ ] Documentation complete
- [ ] Collection variants added

### PR DESCRIPTION
**Pull request by Raffi02k**
### **Description:**  
Implemented the `transpose` method in the `Arrays` utility class to transpose 2D arrays as required by the project:  

- **`transpose(T[][] array)`** - Converts rows into columns and vice versa.  
- Ensures input arrays are **rectangular**, throwing an exception for irregular arrays.  
- Handles **empty arrays** gracefully.  
- Preserves **type information** using `Array.newInstance()`.  
- Supports **arrays with `null` values** without errors.  
- Optimized for **performance**, successfully handling **large arrays** (e.g., 1000x1000).  

### **Updated Unit Tests:**  
- **Square arrays** - Verified correct transposition.  
- **Rectangular arrays** - Ensured proper handling of non-square matrices.  
- **Single row/column** - Confirmed expected transformation.  
- **Empty arrays** - Validated correct handling.  
- **Irregular arrays** - Throws `IllegalArgumentException` as expected.  
- **Large arrays** - Performance tested.  
- **Arrays with `null` elements** - Ensured correct behavior.  
- **Type preservation** - Verified using different generic types.  

### **Changes:**  
- Implemented `transpose` method in `Arrays` class.  
- Added comprehensive unit tests for all edge cases.  

### **Issue Closed:**  
This PR **closes #22** – **Implement array transpose operation in Arrays utility class.** 🚀

## Summary by CodeRabbit

- **New Features**
	- Introduced a new utility for transposing two-dimensional arrays with built-in validations.
- **Tests**
	- Added comprehensive tests for the transposition functionality, covering various scenarios including square, rectangular, single row/column, irregular arrays, large arrays, and edge cases involving null or empty arrays.

### CODE REVIEW

File: src/main/java/org/fungover/breeze/util/Arrays.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package org.fungover.breeze.util;
+
+import java.lang.reflect.Array;
+
+public class Arrays {
+    public static <T> T[][] transpose(T[][] array) {
+        //Checks if the input array is null or if it has no rows or columns
+        if (array == null || array.length == 0 || array[0].length == 0) {
```
Comment (ArdaHalim2) [inline src/main/java/org/fungover/breeze/util/Arrays.java:8]: Looks good! Maybe a risk of ArrayOutOfBoundsIndex if the array is empty

**Review by ArdaHalim2 — CHANGES_REQUESTED** [review]
Looks good! Maybe a risk of ArrayOutOfBoundsIndex if the array is empty

**Comment by liamr03** [comment]
@ArdaHalim2 

> “Live as if you were to die tomorrow. Learn as if you were to live forever.” - Raffi2k

File: src/main/java/org/fungover/breeze/util/Arrays.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            return array;
+        }
+
+        //If the input array is valid, it initializes rows and cols to the dimensions of the array.
+        int rows = array.length;
+        int cols = array[0].length;
+
+        //Validates that all rows in the array have the same length.
```
Comment (ArdaHalim2) [inline src/main/java/org/fungover/breeze/util/Arrays.java:16]: Docoment the code more thoroughly by explaining what parameters and inputs/throws the code is using.

for example:

`/**
 * Validates that a 2D array is well-formed by checking that:
 * - No row is null.
 * - All rows have the same length.
 *
 * @param array The 2D array to validate.
 * @param cols  The expected number of columns in each row.
 * @param <T>   The type of elements in the array.
 * @throws IllegalArgumentException if a row is null or if row lengths are inconsistent.
 */`

**Review by ArdaHalim2 — CHANGES_REQUESTED** [review]
Looks good! Just fix the documentations!

> The universe is not fair and it is never going to be fair. -Nautica Malone

**Review by ArianG421 — CHANGES_REQUESTED** [review]

**Review by ArianG421 — DISMISSED** [review]

**Review by ArdaHalim2 — APPROVED** [review]

**Review by ArianG421 — APPROVED** [review]
Looks good, well done cleaning up.
